File: solutions/003/003.js
function largestPrimeFactor(number) {
  let factor = 1
  let result = number
  for (let i = 3; i * i <= result; i += 2) {
    if (result === 1)
      return factor
    else if (result % i !== 0)
      continue
    else {
      factor = i
      while (result % i === 0)
        result /= i
    }
  }
  return result
}

console.log(largestPrimeFactor(600851475143))

File: solutions/003/003.c
#include <stdio.h>

long largestPrimeFactor(long number) {
        long factor = 1;
        for (int i = 3; i * i <= number; i += 2) {
            if (number == 1)
                return factor;
            else if (number % i != 0)
                continue;
            else {
                factor = i;
                while (number % i == 0)
                    number /= i;
            }
        }
        return number;
}

int main() {
   printf("%ld\n", largestPrimeFactor(600851475143));
}
